live-build/misc/upgrade-scripts/verify-impl

```diff
@@ -44,6 +44,81 @@ function cleanup() {
 	return "$rc"
 }
 
+function verify_run_verify_jar() {
+	echo "Verify - Run verify jar."
+
+	/usr/bin/java \
+		-Dlog.dir=/var/delphix/server/upgrade-verify \
+		-Dmdsverify=true \
+		"$VERIFY_DEBUG_OPT" \
+		-jar /opt/delphix/server/lib/exec/upgrade-verify/upgrade-verify.jar \
+		-d "${opt_o:-$DROPBOX_DIR/upgrade_verify_report.json}" \
+		-f "${opt_f:-1}" \
+		-l "${opt_l:-en-US}" \
+		-v "$IMAGE_VERSION" \
+		-pl 60 -ph 80 \
+		"$VERIFY_LIVE_MDS_OPT" ||
+		die "'upgrade-verify.jar' failed in verification container"
+
+	#
+	# This name is used by the "upgrade-verify.jar" just executed, so we
+	# cannot change this value without also modifying that JAR.
+	#
+	MDS_SNAPNAME="MDS-CLONE-upgradeverify"
+
+	/opt/delphix/server/bin/dx_manage_pg stop -s "$MDS_SNAPNAME" ||
+		die "failed to stop postgres"
+
+	/opt/delphix/server/bin/dx_manage_pg cleanup -s "$MDS_SNAPNAME" ||
+		die "failed to cleanup postgres"
+}
+
+function verify_stack_startup() {
+	echo "Verify - Run stack startup."
+
+	#
+	# Make sure delphix-mgmt and delphix-postgres aren't running within container.
+	#
+	[[ $(systemctl is-active delphix-mgmt) == 'inactive' ]] ||
+		die "delphix-mgmt service is not inactive."
+	[[ $(systemctl is-active delphix-postgres@default.service) == 'inactive' ]] ||
+		die "delphix-postgres@default.service service is not inactive."
+
+	#
+	# Edit Postgres systemd.service file to run within container.
+	#
+	local delphix_postgres_service='/lib/systemd/system/delphix-postgres@.service'
+	sed -i 's/ConditionVirtualization/# &/' ${delphix_postgres_service}
```

Comment:
We should use `|| die` here.

Would it be possible to incorporate all of this MDS manipulation logic into the postgres service, itself? E.g. have the service detect that it's running in a container, then clone/copy the data, then start itself using that clone? This we could just remove this condition from the service's definition, and let it start up while in the container.

It just seems awkward to do this in the verification script, when it's dependent on the internal workings of the service, and is prone to break if/when the service changes.

I'm not necessarily against doing it this way, but I'd be in favor of coming up with a better solution that didn't expose so much of the app-gate service's internals. Ideally, I think we could just rely on the service starting correctly when it's in the container, and then pointing the app-gate stack at this "containerized" postgres instance.

If you'd rather go with this for now, I'm OK with that, but I'd like to at least open a bug to investigate if we could reduce the coupling between the service and this script.

Another thing that I'm worried about, is which port will this postgres instance use? Presumably it'll use the default port, which is the same as the "production" MDS instance? If so, I think this will work now, because the container is configured to have a separate/isolated network stack than the host. I'm not sure we want to do this long term, though, since I think the container will need to be able to reach hosts external to the DE (e.g. so it can reach target environments). So we might want to be proactive, and ensure this second MDS instance uses a non-default port, in anticipation of the container sharing the network with the host.

Reply:
> Would it be possible to incorporate all of this MDS manipulation logic into the postgres service, itself?
New revision leverages `dx_manage_pg.sh` script to manipulate a copy of MDS. This script internally creates snapshot, systemd service definition, starts, stops postgres internally.

> Another thing that I'm worried about, is which port will this postgres instance use?
Ah. I haven't though about this, but I'm hard-coding `dx_manage_pg.sh` script to use default port of 5432. I agree that it'll likely be desirable that we want to start app-stack with custom postgres port but It seems that it's set during the build phase. I've filed https://jira.delphix.com/browse/DLPX-62625 to track this work.